ddtrace/internal/utils/http.py

```diff
@@ -158,7 +158,12 @@ def w3c_get_dd_list_member(context):
         tags.append("{}:{}".format(W3C_TRACESTATE_SAMPLING_PRIORITY_KEY, context.sampling_priority))
     if context.dd_origin:
         # the origin value has specific values that are allowed.
-        tags.append("{}:{}".format(W3C_TRACESTATE_ORIGIN_KEY, re.sub(r",|;|=|[^\x20-\x7E]+", "_", context.dd_origin)))
+        tags.append(
+            "{}:{}".format(
+                W3C_TRACESTATE_ORIGIN_KEY, re.sub(r",|;|~|[^\x20-\x7E]+", "_", context.dd_origin).replace("=", "~")
```

Comment:
I wonder if we can just add `~_=~` into the regex itself instead of doing a replacement outside the regex operation? Feels like it would be cleaner this way unless these replacements have nothing to do with the regular expression.

Reply:
I couldn't figure out how to do this without using lambda, which I don't think is supported in all Python versions we support? However, if you know there's a way, I agree, it would be cleaner.

Reply:
I think breaking this out into a separate function makes it cleaner.